ToolkitApp/tabs/src/components/i18n.js

```diff
@@ -0,0 +1,122 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+
+import Backend from 'i18next-http-backend';
+import LanguageDetector from 'i18next-browser-languagedetector';
+
+import enUS from '../translations/en-US.json';
+import deDE from '../translations/de-DE.json';
+import esES from '../translations/es-ES.json';
+import frFR from '../translations/fr-FR.json';
+import jaJP from '../translations/ja-JP.json';
+import ptBR from '../translations/pt-BR.json';
+import ruRU from '../translations/ru-RU.json';
+import zhCN from '../translations/zh-CN.json';
+
+import "@microsoft/teams-js";
+
+const fallbackLang = 'en';
+
+const supportedLangs = [
+  'en',
+  'de',
+  'es',
+  'fr',
+  'ja',
+  'pt',
+  'ru',
+  'zh',
+  'en-US',
+  'de-DE',
+  'es-ES',
+  'fr-FR',
+  'ja-JP',
+  'pt-BR',
+  'ru-RU',
+  'zh-CN', 
+];
+
+// We need to abstract the language changing because we want to:
+// 1. Process the locale code
+// 2. Make sure that we do not trigger an update if the language is already set to the processed code
+export const changeLanguage = (i18n, code) => {
+  let locale = processLanguageCode(code);
+  if (i18n.languages) {
+    if (i18n.languages[0] === locale) {
+      return;
+    }
+  }
+  i18n.changeLanguage(locale);
+}
+
+const processLanguageCode = (code) => {
+  if (!code) {
```

Comment:
Will this work? I feel we don't need two `supportedLangs.find` here : >

```
const processLanguageCode = (code) => {
  let langCode = code.toLowerCase();

  if (langCode.includes("_")) {
    langCode = langCode.split("_")[0];
  }

  let supported = supportedLangs.find(l => l.toLowerCase().startsWith(langCode));

  return supported || fallbackLang;
}
```

Reply:
The two supported lang finds simply enables the localization to prioritize country-specific locales first, and then default to the language-specific locale. I’m not sure that I want to remove this feature because I think it is useful for if we add future locale files like “en-CA.json” in the future; Keeping the option open seems more scalable.

Reply:
got it. But I guess we still don't need 2 passes of the list. We could do 1 pass but check 2 versions of locale.